### PR TITLE
support labelselector set

### PR DESCRIPTION
**Pull request by ldpliu**
Signed-off-by: ldpliu <daliu@redhat.com>

### CODE REVIEW

File: pkg/controllers/scheduling/scheduling_controller.go [hunk 1]
```diff
@@ -352,58 +352,56 @@ func (c *schedulingController) getAvailableClusters(clusterSetNames []string) ([
...
-			continue
-		}
-
-		labelSelector := labels.NewSelector().Add(*requirement)
-		clusters, err := c.clusterLister.List(labelSelector)
+		clusters, err := c.getClusterSetClusters(clusterSet)
 		if err != nil {
-			klog.Warning(err)
-			continue
+			klog.Warning("Failed to get clusterset: %v, clusters", clusterSet.Name, err)
```
Comment (qiujian16) [inline pkg/controllers/scheduling/scheduling_controller.go:366]: @haoqing0110 I think we should return error in the func and reflect in the condition?

**Review by qiujian16 — COMMENTED** [review]
/approve

one nit comment, I am not sure how we let user know that clusterset has not clusters.

File: pkg/controllers/scheduling/scheduling_controller.go (hunk 1 above)
Comment (haoqing0110) [reply pkg/controllers/scheduling/scheduling_controller.go:366]: @qiujian16 yes, PR https://github.com/open-cluster-management-io/placement/pull/72 is in progress, getAvailableClusters could return error or framework.Status (in the future) to indicate different kinds of errors and reflect in the condition.

**Comment by ldpliu** [comment]
@elgnay Please help to review this pr.
Thank you

**Pull request merged**